src/gluonts/dataset/dataframe.py

```diff
@@ -68,9 +69,10 @@ class DataFramesDataset(Dataset):
     dataframes: Union[
         List[pd.DataFrame], Dict[str, pd.DataFrame], pd.DataFrame
     ]
-    target: Union[str, List[str]]
-    timestamp: str
-    freq: str
+    target: Union[str, List[str]] = "target"
+    timestamp: str = ""
+    freq: str = ""
+    item_id: str = ""
```

Comment:
Should these be `None` instead by default?

Reply:
I had them initially as `None`s. But then the type checker complains about the None type and I have to manually cast (typing.cast) all occurrences of `timestamp`, `freq` and `item_id` in all methods to `str`. Do you want to go for `None`? For me both are fine.

Reply:
`Union[str, None]` would be the input to the initializer. In the initializer they are set to `str`. So, in other methods of the class we can assume their types to be `str`. Now,  the input to a method is of type `Optional[str]` (inferred by the checker) but the method only accepts `str`. This is why the checker complains.
Here, e.g., `item_id` is of type `str` but in init it would be `Union[str, None]`:
```python
def _dataentry(self, item_id: str, df: pd.DataFrame) -> DataEntry:
```
This seems to be in general a problem of the `Union` type :shrug:.

Reply:
I think the issue is that `dataclass` mixes the type of the passed argument with the object's attribute.

I guess we can either use an alternative constructor or not use a `dataclass`.

Reply:
@RSNirwan maybe there are some other issues, but in the `_dataentry` esample you mentioned, I guess `_dataentry` should just accept `item_id: Optional[str]` and it would be fine